shopfloor_checkout_putinpack_restriction/services/checkout.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+
+from odoo.addons.component.core import Component
+
+
+class Checkout(Component):
+    _inherit = "shopfloor.checkout"
+
+    def _data_response_for_select_package(self, picking, lines, message=None):
+        res = super()._data_response_for_select_package(picking, lines)
+        if picking.put_in_pack_restriction == "no_package":
+            res["package_allowed"] = False
+        elif picking.put_in_pack_restriction == "with_package":
+            res["no_package_enabled"] = False
```

Comment:
I would make the rule clear here, without relying on the default value of `shopfloor`.
```python
        if picking.put_in_pack_restriction == "no_package":
            res["package_allowed"] = False
            res["no_package_enabled"] = True
        elif picking.put_in_pack_restriction == "with_package":
            res["package_allowed"] = True
            res["no_package_enabled"] = False
```
And when we'll remove the `no_package_enabled` option we won't need to think too much.